File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    ret = []
    keyDict = {}
    
    for i in range(len(keymap)):
        for j in range(len(keymap[i])):
            ch = keymap[i][j]
            if ch in keyDict.keys():
                keyDict[ch] = min(keyDict[ch], j+1)
            else:
                keyDict[ch] = j+1
                
    for target in targets:
        cnt = 0
        for ch in target:
            if ch in keyDict.keys():
                cnt += keyDict[ch]
            else:
                cnt = -1
                break
                
        ret.append(cnt)
        
    return ret

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#include <map>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    map<char, int> mp;
    vector<int> ret;

    for (int i = 0; i < keymap.size(); i++) {
        for (int j = 0; j < keymap[i].size(); j++) {
            char c = keymap[i][j];
            if (mp.count(c) == 0) mp[c] = j + 1;
            else mp[c] = min(mp[c], j + 1);
        }
    }

    for (string target: targets) {
        int sum = 0;

        for (char c: target) {
            if (mp.count(c) == 0) {
                sum = -1;
                break;
            }
            sum += mp[c];
        }
        ret.push_back(sum);
    }
    return ret;
}
